specification/botservice/resource-manager/Microsoft.BotService/stable/2021-03-01/botservice.json

```diff
@@ -0,0 +1,2640 @@
+          "type": "string"
+        },
+        "value": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ServiceProvider"
+          },
+          "description": "Gets the list of bot service providers and their properties."
+        }
+      },
+      "description": "The list of bot service providers response."
+    },
+    "ServiceProviderParameter": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Name of the Service Provider",
+          "readOnly": true
+        },
+        "type": {
+          "type": "string",
+          "description": "Type of the Service Provider",
+          "readOnly": true
+        },
+        "displayName": {
+          "type": "string",
+          "description": "Display Name of the Service Provider",
+          "readOnly": true
+        },
+        "description": {
+          "type": "string",
+          "description": "Description of the Service Provider",
+          "readOnly": true
+        },
+        "helpUrl": {
+          "type": "string",
+          "description": "Help Url for the  Service Provider",
+          "readOnly": true
+        },
+        "default": {
+          "type": "string",
+          "description": "Default Name for the Service Provider",
+          "readOnly": true
+        }
+      },
+      "description": "Extra Parameters specific to each Service Provider"
+    },
+    "ServiceProviderProperties": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Id for Service Provider",
+          "readOnly": true
+        },
+        "displayName": {
+          "type": "string",
+          "description": "Display Name of the Service Provider",
+          "readOnly": true
+        },
+        "serviceProviderName": {
+          "type": "string",
+          "description": "Display Name of the Service Provider",
+          "readOnly": true
+        },
+        "devPortalUrl": {
+          "type": "string",
+          "description": "Display Name of the Service Provider",
+          "readOnly": true
+        },
+        "iconUrl": {
+          "type": "string",
+          "description": "Display Name of the Service Provider",
+          "readOnly": true
+        },
+        "parameters": {
+          "description": "The list of parameters for the Service Provider",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ServiceProviderParameter"
+          }
+        }
+      },
+      "description": "The Object used to describe a Service Provider supported by Bot Service"
+    },
+    "ServiceProvider": {
+      "type": "object",
+      "description": "Service Provider Definition",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ServiceProviderProperties",
+          "description": "The Properties of a Service Provider Object"
+        }
+      }
+    },
+    "Error": {
```

Comment:
ideally, you would reference the common error type, but we do not want breaking changes

Reply:
existing schema